lib.go

```diff
@@ -1126,3 +1128,83 @@ func truncateSheetXML(r io.Reader, rowLimit int) (io.Reader, error) {
 	}
 	return output, nil
 }
+
+// truncateSheetXMLValueOnly will take in a reader to an XML sheet file and will return a reader that will read an equivalent
+// XML sheet file without null vaules of rows. This greatly speeds up XML unmarshalling when we
+// only need non-NULL data for the sheet.
+// When sheets are truncated, most of formatting present will be not right, but all of this formatting
+// is related to printing and visibility, and is out of scope for most purposes of this library.
+func truncateSheetXMLValueOnly(r io.Reader) (io.Reader, error) {
```

Comment:
Though I understand the caveats around the meaning of NULL, I really like your approach to this.  Preprocessing the XML whilst maintaining the Reader interface is a smart hack.  👍🏻